packages/core/lib/command-utils.js

```diff
@@ -206,9 +212,85 @@ const displayGeneralHelp = () => {
     .showHelp();
 };
 
+const getConfiguredNetworkUrl = function (customConfig, isDashboardNetwork) {
+  const defaultPort = isDashboardNetwork
+    ? managedDashboardDefaultPort
+    : managedGanacheDefaultPort;
+  const configuredNetworkOptions = {
+    host: customConfig.host || defaultHost,
+    port: customConfig.port || defaultPort
+  };
+  const urlSuffix = isDashboardNetwork ? "/rpc" : "";
+  return `http://${configuredNetworkOptions.host}:${configuredNetworkOptions.port}${urlSuffix}`;
+};
+
+const deriveConfigEnvironment = function (detectedConfig, network, url) {
+  let configuredNetwork;
+
+  const configDefinesProvider =
+    detectedConfig.networks[network] &&
+    detectedConfig.networks[network].provider;
+
+  if (configDefinesProvider) {
+    // Use "provider" specified in the config to connect to the network
+    // along with the other specified network properties
+    configuredNetwork = {
+      ...detectedConfig.networks[network],
+      network_id: "*",
+      provider: detectedConfig.networks[network].provider
+    };
+  } else if (url) {
+    // Use "url" to configure network (implies not "develop" and not "dashboard")
+    configuredNetwork = {
+      network_id: "*",
+      url,
+      provider: function () {
+        return new Web3.providers.HttpProvider(url, {
+          keepAlive: false
+        });
+      }
+    };
+  } else {
+    // Otherwise derive network settings
+    const customConfig = detectedConfig.networks[network] || {};
+    const isDashboardNetwork = network === "dashboard";
+    const configuredNetworkUrl = getConfiguredNetworkUrl(
+      customConfig,
+      isDashboardNetwork
+    );
+    const defaultPort = isDashboardNetwork
+      ? managedDashboardDefaultPort
+      : managedGanacheDefaultPort;
+    const defaultNetworkId = isDashboardNetwork
+      ? "*"
+      : managedGanacheDefaultNetworkId;
+
+    configuredNetwork = {
+      // customConfig will spread only when it is defined and ignored when undefined
+      ...customConfig,
+      host: customConfig.host || defaultHost,
+      port: customConfig.port || defaultPort,
+      network_id: customConfig.network_id || defaultNetworkId,
+      provider: function () {
```

Comment:
I'm not sure it makes sense to specify both `host` and `url` along with `provider`. Shouldn't we only specify one or the other?

Also, if we spread the user's settings in here there may be a situation where we end up with both. In that case, what happens? Does one take precedence over the other?

Reply:
In this case,`provider` takes precedence over the other. Because our logic says that if there is a `provider` specified in the config then we use the `provider` to connect to the network and ignore `host` and `port` (The first `if` block of the main function will be executed). Does that make sense?

Reply:
Yes it does! But that seems to mean that we only want to add a provider here in the case that there is one since `host` and `port` will be ignored **every time** since a provider is always specified here, right?

Reply:
That makes sense to me! We can remove the `host` and `port` settings as the provider will always be here. Thanks for catching this @eggplantzzz.